public/app/features/explore/QueryRow.tsx

```diff
@@ -155,8 +163,20 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
                 exploreMode={mode}
                 exploreId={exploreId}
               />
-            ) : (
+            )}
+            {!ExploreQueryField && QueryEditor && (
```

Comment:
Can we encapsulate this decision making a bit more. Right now there are 2 section where we decide what to render `setExploreQueryField` method and then here in render dispersed with the markup. I think we can refactor it so there is one place where the decision is made.